erts/emulator/beam/erl_term.h

```diff
@@ -1294,14 +1296,19 @@ _ET_DECLARE_CHECKED(struct erl_node_*,external_ref_node,Eterm)
      (hp)[1] = sz,                              \
      (hp)[2] = keys)
 
-#define MAP_SZ(sz) (MAP_HEADER_FLATMAP_SZ + 2*sz + 1)
+/* NB. When sz is 0, TUPLE0 is shared, so takes no space */
+#define MAP_SZ(sz) (MAP_HEADER_FLATMAP_SZ + 2*sz + !!sz)
 
+#define MAP0_SZ MAP_SZ(0)
 #define MAP1_SZ MAP_SZ(1)
 #define MAP2_SZ MAP_SZ(2)
 #define MAP3_SZ MAP_SZ(3)
 #define MAP4_SZ MAP_SZ(4)
 #define MAP5_SZ MAP_SZ(5)
 
+#define MAP0(hp)                                                        \
+    (MAP_HEADER(hp, 0, TUPLE0),                    \
+     make_flatmap(hp))
```

Comment:
This ought to be a global literal just like `TUPLE0`.

Reply:
I agree, but iiuc this is not something I can do as part of this change, correct? The deduplication needs to happen somewhere, etc